Tools/TestWebKitAPI/Tests/IPC/EventTests.cpp

```diff
@@ -83,7 +83,6 @@ TEST_P(EventTestABBA, SerializeAndSignal)
         ASSERT_TRUE(openB());
 
         bClient().setAsyncMessageHandler([&] (IPC::Decoder& decoder) -> bool {
-            decoder.decode<uint64_t>();
```

Comment:
This decoding was always failing. However, we were not checking if it succeeded or not. Despite failing, we would successfully decode the Signal on the next line with the old decoder.

However, the new generated code does:
```
auto signal = decoder.decode<IPC::Signal>(); // Succeeds and does NOT return nullopt.
if (!decoder.isValid())
    return std::nullopt; // We return here because the decoder was marked as invalid by the earlier decoding of the uint64_t.
```

Reply:
I have no explanations for why it was trying to decode an int before the actual signal.